File: Medium/Javascript/189. Rotate Arrayp.js
/**
 * Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.
 */

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
  if (k > 0) {
    if (k > nums.length) {
      let ind = 0;
      while (ind < k) {
        nums.unshift(nums[nums.length - 1]);
        nums.pop();
        ind++
      }
    }
    else{
        nums.unshift(...nums.slice(nums.length - k))
        nums.splice(nums.length - k,k)
    }
  }

  return nums
};

console.log(rotate([1,2,3,4,5,6,7],3))

File: Medium/python/189. Rotate Arrayp.py
'''
Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.
'''

class Solution:
    def rotate(self, nums: list[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)

        if k > 0:
            if k < length:
                nums[:] = nums[length - k:] + nums[0: length - k]
            else:
                ind = 1
                while ind <= k:
                    nums[:] = nums[-1:] + nums[0:-1]
                    ind += 1
